### DISCUSSION TITLE
how can I add an existing C# project to a RNW solution?

### DISCUSSION THREAD

**Discussion by BenMakesGames** [microsoft/react-native-windows#8410] [Q&A] [answered]
**context:** I'm looking for an alternative to WPF for making a desktop application. a lot of code has already been written in .NET 5 C# for this application: service classes, entity framework, etc. conceptually, I'm looking to replace the UI layer, swapping out WPF for React Native. (I'm aware it will probably be a little more difficult than just that, but the project is small still; as a point of comparison, I've also been investigating Blazor Desktop.)

**question:** what's the recommended/best/fastest way to take existing .NET 5 C# business logic classes (services, dbcontexts...), and use them in an RNW application? or is this a fool's errand?

(maybe worth adding: .NET 5 probably isn't a requirement; we'd like to move to .NET 6 when it's out, since that will be an LTS release.)

**things I have already tried:**
* I found https://microsoft.github.io/react-native-windows/docs/native-modules and looked through that... it wasn't clear to me, from those instructions, how to use an existing project. (make a new project, and copy-paste all the C# files over?)
* I tried simply adding our existing database project to the RNW solution (in the windows/ dir), but when I attempted to add that project as a reference to the RNW UWP project, Visual Studio threw an error message ("Unable to add a reference to project 'DataAccess'"), lagged out for a while, and then threw another ("The method or operation is not implemented."??? huh.)
   * I tried changing the TargetFramework of the existing project from net5.0 to netcoreapp3.1, and received the same errors; I tried netcoreapp2.2 as well, and received the same errors

**Comment by chrisglein** [comment, marked as answer]
If it helps, [this article](https://techcommunity.microsoft.com/t5/windows-dev-appconsult/getting-started-with-react-native-for-windows/ba-p/912093) covers how to host RNW inside a WPF application. Which isn't quite what you're asking for, but maybe relevant? It certainly comes from more of an established app perspective, not just a new project.

One key difference coming from WPF to RNW is that RNW is going to create a UWP app, which is going to affect the portability of some of your existing app logic. More on that [here](https://microsoft.github.io/react-native-windows/docs/native-code#win32-desktop-apps-vs-rnw-apps).

**Comment by asklar** [comment]
on the topic of UWP vs. .NET5, UWP apps do not support .net 5. The currently-experimental way to use .net 5 is to use WinUI3 and the WindowsAppSdk. This is still work in progress.
